content/Training/developers/04-agent-collaboration.mdx

```diff
@@ -0,0 +1,734 @@
+            if isinstance(item, dict):
+                title = item.get("title", "")
+                snippet = item.get("snippet", "")
+                url = item.get("url", "")
+                relevance = item.get("relevance_score", 0)
+                summary_parts.append({
+                    "rank": i + 1,
+                    "title": title,
+                    "summary": snippet,
+                    "source_url": url,
+                    "relevance": relevance
+                })
+
+        final_summary = {
+            "query": original_query,
+            "summary": f"Research Summary for '{original_query}'",
+            "key_findings": summary_parts,
+            "total_sources": len(content),
+            "research_completed_at": datetime.now().isoformat(),
+            "agent_chain": ["orchestrator", "web-search", "summarizer"]
+        }
+    else:
+        # Handle direct text content
+        final_summary = {
+            "query": original_query,
+            "summary": f"Summary of content for '{original_query}'",
+            "content_length": len(str(content)),
+            "processed_at": datetime.now().isoformat(),
+            "agent_chain": ["orchestrator", "summarizer"]
+        }
+
+    # Return the final research summary to the user
+    return response.json(final_summary)`} js={`const handler: AgentHandler = async (request, response, context) => {
+  // Specialist agent for summarizing research content
+
+  const summaryRequest = await request.data.json();
+  const content = summaryRequest.content || [];
+  const originalQuery = summaryRequest.original_query || '';
+  const source = summaryRequest.source || 'unknown';
+
+  context.logger.info(\`Summarizing \${Array.isArray(content) ? content.length : 1} items for query: \${originalQuery}\`);
+
+  // Process different content types
+  let finalSummary;
+
+  if (source === 'web_search' && Array.isArray(content)) {
+    // Handle web search results
+    const summaryParts = content.slice(0, 3).map((item, i) => ({
+      rank: i + 1,
+      title: item.title || '',
+      summary: item.snippet || '',
+      source_url: item.url || '',
+      relevance: item.relevance_score || 0
+    }));
+
+    finalSummary = {
+      query: originalQuery,
+      summary: \`Research Summary for '\${originalQuery}'\`,
+      key_findings: summaryParts,
+      total_sources: content.length,
+      research_completed_at: new Date().toISOString(),
+      agent_chain: ['orchestrator', 'web-search', 'summarizer']
+    };
+  } else {
+    // Handle direct text content
+    finalSummary = {
+      query: originalQuery,
+      summary: \`Summary of content for '\${originalQuery}'\`,
+      content_length: String(content).length,
+      processed_at: new Date().toISOString(),
+      agent_chain: ['orchestrator', 'summarizer']
+    };
+  }
+
+  // Return the final research summary to the user
+  return response.json(finalSummary);
+};`} />
+
+## Testing Your Research Orchestrator
```

Comment:
Why not lean in to dev mode as the first way to test it out, and mention curl as a programmatic alternative?